main.go

```diff
@@ -126,39 +122,71 @@ func (e *expander) expandEnviron(decrypt bool) error {
 		}
 
 		if parameter != nil {
-			names[*parameter] = true
+			uniqNames[*parameter] = true
 			ssmVars = append(ssmVars, ssmVar{k, *parameter})
 		}
 	}
 
-	for k := range names {
-		input.Names = append(input.Names, aws.String(k))
-	}
-
-	if len(input.Names) == 0 {
+	if len(uniqNames) == 0 {
 		// Nothing to do, no SSM parameters.
 		return nil
 	}
 
+	names := make([]string, len(uniqNames))
+	i := 0
+	for k := range uniqNames {
+		names[i] = k
+		i++
+	}
+
+	batchSize := 10
```

Comment:
Can you expand the tests to include one that exercises batching? The implementation here looks correct, but it doesn't look like all the code paths are touched.

Reply:
No problem, I should have done that to begin with 😬 
I've updated it with a test that should test two batches of size one.